multistream-select/src/protocol/listener.rs

```diff
@@ -113,11 +113,13 @@ impl<R> Sink for Listener<R>
 			}
 
 			ListenerToDialerMessage::ProtocolsListResponse { list } => {
+				use std::iter;
```

Comment:
I'm not sure what the best practice is, but I usually prefer to put everything at the top of the file.

Reply:
I try and make the scope the minimum possible without unnecessary duplication, but I can't find any consensus online on local imports. I don't think we `use std::iter` anywhere else in the file so I think it's fine here.

Reply:
I agreed with this in the past, but the problem is that I noticed that over time you tend to put the same local import(s) at multiple local scopes simultaneously, and things turn into spaghetti.

Reply:
Fair enough, I'll change it